forum/qa-plugin/user-activity-log/user-activity-log-search.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+class user_activity_search
+{
+    
+    private $searchArr;
+    private $dbResult;
+
+    public function match_request($request) 
+    {
+        $this->searchArr = $_POST;
+        isset($_POST['request']) ? $_SESSION['query'] = $_POST['request'] : null;
+        isset($_POST['date']) ? $_SESSION['date'] = $_POST['date'] : null;
+        isset($_POST['resultsCount']) ? $_SESSION['resultsCount'] = $_POST['resultsCount'] : null;
+        isset($_POST['sortBy']) ? $_SESSION['sortBy'] = $_POST['fromOldest'] : null;
+
+        if(!isset($_POST['condition']) || !isset($_POST['resultsCount'])){
+            header('Location: ../../');
+            exit;
+        }
+
+        if(qa_get_logged_in_level() >= QA_USER_LEVEL_EDITOR){
+            if(isset($this->searchArr)){
+                $this->dbSearch();
+                if(isset($this->dbResult)){
+                    return $request === 'user-activity-log-search';
+                }
+            }
+        }else{
+            return false;
+        }
+    }
+
+    public function process_request($request)
+    {
+        $qa_content=qa_content_prepare();
+        $qa_content['title']= '<a class = "to-right" href = "'.qa_path_to_root().'user-activity-log">'
+            .qa_lang_html('user-activity-log/searchResults')
+            .$this->searchArr['request'].'</a>';
+            
+        $qa_content['custom'] = $this->generateResultsTable();
+        $qa_content['navigation']['sub']=qa_admin_sub_navigation();
+        return $qa_content;
+    }
+
+    private function dbSearch()
+    {
+        $baseQuery = 'SELECT `datetime`, `ipaddress`, `handle`, `event`,`params` FROM qa_eventlog ';
+
+        if(!empty($this->searchArr['request'])){
+            if($this->searchArr['condition'] === 'username'){
+                $finalQuery = $baseQuery.'WHERE handle = $ ';
+            }else if($this->searchArr['condition'] === 'type'){
+                $finalQuery = $baseQuery.'WHERE event = $ ';
+            }else if($this->searchArr['condition'] === 'ip'){
+                if(qa_get_logged_in_level() > QA_USER_LEVEL_EDITOR){
+                    $finalQuery = $baseQuery.'WHERE ipaddress = $ ';
+                }   
+            }
+        }else{
+            $finalQuery = $baseQuery;
+        }
+
+        if(!empty($this->searchArr['date'])){
+            $dateArr = explode('-', $this->searchArr['date']);
```

Comment:
Możesz tu zamiast tworzyć `$dateArr` to:
```php
[$day, $month, $year] = explode('-', $date);
```
I używać bardziej czytelnych zmiennych niż niżej.

Reply:
Tutaj nie zawsze wiemy jak dokładnie data będzie podana. Może to być coś w stylu "2022-02-10 09" I wtedy (używając twojego sposobu) nam się  dobrze uruchomi. Ale gdy wpiszemy np. "2022" to już dostaniemy Undefined offset zamiast wyniku.